src/sentry/incidents/endpoints/validators.py

```diff
@@ -103,7 +109,85 @@ def validate(self, attrs):
+            updated_values = {
+                "type": data_conditions.get("type", data_condition.type),
+                "comparison": data_conditions.get("comparison", data_condition.comparison),
+                "condition_result": data_conditions.get("result", data_condition.condition_result),
+            }
+            data_condition.update(**updated_values)
+            return instance.workflow_condition_group
+
+        condition_group = DataConditionGroup.objects.create(
+            logic_type=DataConditionGroup.Type.ANY,
+            organization_id=self.context["organization"].id,
+        )
+        DataCondition.objects.create(
+            type=data_condition.get("type"),
+            comparison=data_condition.get("comparison"),
+            condition_result=data_condition.get("result"),
+            condition_group=condition_group,
+        )
+        return condition_group
+
+    def update_data_source(self, instance, data_source):
+        for source in data_source:
+            try:
+                source_instance = DataSource.objects.get(detector=instance)
+            except DataSource.DoesNotExist:
+                continue
+            if source_instance:
+                try:
+                    snuba_query = SnubaQuery.objects.get(id=source_instance.query_id)
+                except SnubaQuery.DoesNotExist:
+                    raise serializers.ValidationError("SnubaQuery not found, can't update")
+
+            event_types = SnubaQueryEventType.objects.filter(snuba_query_id=snuba_query.id)
+            update_snuba_query(
+                snuba_query=snuba_query,
+                query_type=source.get("query_type", snuba_query.type),
+                dataset=source.get("dataset", snuba_query.dataset),
+                query=source.get("query", snuba_query.query),
+                aggregate=source.get("aggregate", snuba_query.aggregate),
+                time_window=source.get("time_window", timedelta(seconds=snuba_query.time_window)),
+                resolution=timedelta(seconds=source.get("resolution", snuba_query.resolution)),
+                environment=source.get("environment", snuba_query.environment),
+                event_types=source.get("event_types", [event_types]),
+            )
+            # TODO handle adding an additional DataSource
+
+    def update(self, instance, validated_data):
+        instance.name = validated_data.get("name", instance.name)
+        instance.type = validated_data.get("group_type", instance.group_type).slug
+        data_conditions = validated_data.pop(
+            "data_conditions"
+        )  # TODO this is not a m2m, should be updated to data_condition singular
+        if data_conditions:
+            instance.workflow_condition_group = self.update_data_condition(
+                instance, data_conditions
+            )
+        # TODO check if DCG logic_type needs to be updated ?
```

Comment:
Would the `DataConditionGroup` logic type ever change when updating the `Detector`?

Reply:
afaik, no, we should hardcode the logic type for detectors to always be `ANY` -- we need to evaluate all the conditions cause it could go from "Ok" -> "low" & "high" at the same time -- it's critical for us to know that multiple conditions are triggered.